examples/seismic/tutorials/06_skimage_tv.ipynb

```diff
@@ -0,0 +1,416 @@
+    "    # it only worked relaiably with Dask master. Therefore, the\n",
+    "    # the model is communicated via a file.\n",
+    "    model0 = get_current_model(param)\n",
+    "    src, rec, nt, solver = get_data(param)\n",
+    "    \n",
+    "    # Create symbols to hold the gradient and the mismatch between\n",
+    "    # the 'measured' and simulated data.\n",
+    "    grad = Function(name=\"grad\", grid=model0.grid)\n",
+    "    residual = Receiver(name='rec', grid=model0.grid, ntime=nt, coordinates=rec.coordinates.data)\n",
+    "    \n",
+    "    # Compute simulated data and full forward wavefield u0\n",
+    "    d, u0, _ = solver.forward(src=src, m=model0.m, save=True)\n",
+    "        \n",
+    "    # Compute the data mismatch (residual) and objective function  \n",
+    "    residual.data[:] = d.data[:] - rec.data[:]\n",
+    "    f = .5*np.linalg.norm(residual.data.flatten())**2\n",
+    "    \n",
+    "    # Compute gradient using the adjoint-state method. Note, this\n",
+    "    # backpropogates the data mismatch through the model.\n",
+    "    solver.gradient(rec=residual, u=u0, m=model0.m, grad=grad)\n",
+    "    \n",
+    "    # Taking here to avoid a (probably overzealous) destructor deleting\n",
+    "    # the gradient before Dask has had a chance to communicate it.\n",
+    "    g = np.array(grad.data[:])\n",
+    "    \n",
+    "    # return the objective functional and gradient.\n",
+    "    return fg_pair(f, g)\n",
+    "\n",
+    "import numpy as np\n",
+    "from distributed import Client\n",
+    "\n",
+    "# Dumps the model to disk; workers will pick this up when they need it.\n",
+    "def dump_model(param, model):\n",
+    "    np.save(param['model'], model.astype(np.float32))\n",
+    "\n",
+    "def fwi_gradient(model, param):\n",
+    "    # Dump a copy of the current model for the workers\n",
+    "    # to pick up when they are ready.\n",
+    "    param['model'] = \"model_0.npy\"\n",
+    "    dump_model(param, model)\n",
+    "\n",
+    "    # Define work list\n",
+    "    work = [dict(param) for i in range(param['nshots'])]\n",
+    "    for i in  range(param['nshots']):\n",
+    "        work[i]['shot_id'] = i\n",
+    "        \n",
+    "    # Start dask cluster\n",
+    "    client = Client()\n",
+    "    \n",
+    "    # Map worklist to cluster\n",
+    "    fgi = client.map(fwi_gradient_i, work)\n",
+    "    \n",
+    "    # Perform reduction on cluster\n",
+    "    fg = client.submit(sum, fgi).result()\n",
+    "    \n",
+    "    # Shutdown cluster\n",
+    "    client.close()\n",
+    "    \n",
+    "    # L-BFGS in scipy expects a flat array in 64-bit floats.\n",
+    "    return fg.f, fg.g.flatten().astype(np.float64)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## FWI with L-BFGS-B\n",
+    "Equipped with a function to calculate the functional and gradient, we are finally ready to define the optimization function."
```

Comment:
"function ... functional ... optimization function." -> used the word function too many times with different meanings.

Reply:
I don't quote agree as function and functional mean two entirely different things. However, would it be clear if I said "python function"?